### PR TITLE
readme(engine): move engine readme to tiflow dir and fix some outdated test discription

### LINKED ISSUE #6853: readme(engine): move tiflow readme to tiflow dir and add some description for e2e test
### Is your feature request related to a problem?

No

### Describe the feature you'd like

1. Move tiflow readme to tiflow dir
2. Add some description for e2e test 
3. Fix some out-dated description for docker deployment

### Describe alternatives you've considered

_No response_

### Teachability, Documentation, Adoption, Migration Strategy

_No response_

### PR DESCRIPTION
**Pull request by maxshuang**
### What problem does this PR solve?

Issue Number: close #6853 

### What is changed and how it works?
1. Move tiflow readme to tiflow dir
2. Split some trivial commands of engine to Makefile.engine
3. Fix some out-dated description for docker deployment

#### Tests 

 - No code

### Release note 

```release-note
 `None`.
```

### CODE REVIEW

File: README_Tiflow.md [hunk 1]
```diff
@@ -4,18 +4,14 @@
 
 This repo implements a new prototype of distributed task scheduler.
```
Comment (lonng) [inline README_Tiflow.md:5]: Still prototype?

File: README_Tiflow.md [hunk 2]
```diff
@@ -4,18 +4,14 @@
...
 
-### Master
+### ServerMaster
```
Comment (lonng) [inline README_Tiflow.md:7]: Why do you change `Master` to `ServerMaster`?

File: README_Tiflow.md (hunk 1 above)
Comment (maxshuang) [reply README_Tiflow.md]: Fix it.

File: README_Tiflow.md (hunk 2 above)
Comment (maxshuang) [reply README_Tiflow.md]: Fix it.

Comment (maxshuang) [reply README_Tiflow.md]: Have we made the decision of replacing `servermaster` with `master`?

File: README_Tiflow.md [hunk 3]
```diff
@@ -93,11 +89,13 @@ If you have build the binary on your local Linux, you can try `../run-engine.sh
 
 ### Deploy
 
-There are several configure files to use. The file name suggests the number of running server-master and executor nodes. For example, `1m1e.yaml` means this file contains one server-master and one executor. Use `../run-engine.sh deploy 1m1e.yaml` to deploy cluster.
+There are several configure files to use. The file name suggests the number of running server-master and executor nodes.   
+For example, `1m1e.yaml` means this file contains one server-master and one executor.   
+Use `../run-engine.sh deploy ./deployments/engine/docker-compose/1m1e.yaml` to deploy cluster.
```
Comment (CharlesCheung96) [inline README_Tiflow.md:94]: Does this command work？

Comment (maxshuang) [reply README_Tiflow.md:94]: image
Sure. The original command fail after yujie move the `engine/deployment`  to `deployments/engine`. 
Just fix it.

File: engine/test/e2e/README.md [hunk 4]
```diff
@@ -0,0 +1,30 @@
+## Preparations
```
Comment (CharlesCheung96) [inline engine/test/e2e/README.md:1]: Better to use debug scripts
```bash
$ configs="./deployments/engine/docker-compose/3m3e.yaml ./deployments/engine/docker-compose/dm_databases.yaml"
$ engine/test/integration_tests/run.sh debug ${configs}

// Now we are in e2e dir
go test -v -race -run="TestDMJob"
```

File: engine/test/e2e/README.md [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+
+The following programs must be installed:
+
+* [docker](https://docs.docker.com/get-docker/)
+* [docker-compose](https://docs.docker.com/compose/install/)
+
+Besides, make sure you have run the docker daemon. We recommend that you provide docker with at least 6+ cores and 8G+ memory. Of course, the more resources, the better.
+
+## Setup environment 
+`NOTICE`: some e2e tests like `e2e_node_chaos_test` rely on the failover mechanism of k8s, so it can't run locally.
```
Comment (CharlesCheung96) [inline engine/test/e2e/README.md:11]: `e2e_node_chaos_test` only rely on docker?

File: engine/test/e2e/README.md (hunk 4 above)
Comment (maxshuang) [reply engine/test/e2e/README.md:1]: Ok. You can give me a suggestion. I will merge it.

**Comment by maxshuang** [comment]
This pr has something conflict with the pr of CharlesCheung96.
Hold it now.

File: README_Tiflow.md (hunk 2 above)
Comment (lonng) [reply README_Tiflow.md]: The `servermaster` is an internal module name and the `master` is a role name. In my opinion, there is no conflict between them.

File: Makefile [hunk 6]
```diff
@@ -1,4 +1,6 @@
 ### Makefile for tiflow
+include Makefile.engine
```
Comment (amyangfei) [inline Makefile:2]: Better to make `include` statement together. L#108

Comment (maxshuang) [reply Makefile]: Fix it.

File: Makefile [hunk 7]
```diff
@@ -481,7 +482,7 @@ failpoint-enable: check_failpoint_ctl
 failpoint-disable: check_failpoint_ctl
 	$(FAILPOINT_DISABLE)
 
-engine: tiflow tiflow-demo
+engine: tiflow tiflow-demo tiflow-chaos-case
```
Comment (amyangfei) [inline Makefile:485]:
```suggestion
engine: tiflow tiflow-demo
```

Comment (amyangfei) [reply Makefile:485]: `chaos-case` is not necessary here

**Review by amyangfei — COMMENTED** [review]
rest lgtm

**Review by amyangfei — APPROVED** [review]

**Review by gozssky — APPROVED** [review]

**Pull request merged**